cache/benches/load_test.rs

```diff
@@ -79,55 +84,55 @@ fn cache_benchmark(c: &mut Criterion) {
 		.noise_threshold(0.03) // def 0.01
 		.measurement_time(Duration::from_secs(15));
 
-	let cache = Cache::new();
+	let cache = Cache::new(zbus_connection.clone());
 	group.bench_function(BenchmarkId::new("add_all", "zbus-docs"), |b| {
-		b.to_async(&rt).iter_batched(
+		b.iter_batched(
 			|| zbus_items.clone(),
 			|items: Vec<CacheItem>| add_all(&cache, items),
 			BatchSize::SmallInput,
 		);
 	});
 
-	let cache = Cache::new();
+	let cache = Cache::new(zbus_connection.clone());
 	group.bench_function(BenchmarkId::new("add", "zbus-docs"), |b| {
-		b.to_async(&rt).iter_batched(
+		b.iter_batched(
 			|| zbus_items.clone(),
 			|items: Vec<CacheItem>| add(&cache, items),
 			BatchSize::SmallInput,
 		);
 	});
 
-	let (cache, children): (Cache, Vec<AccessibleId>) = rt.block_on(async {
+	let (cache, children): (Cache, Vec<AccessiblePrimitive>) = rt.block_on(async {
 		let all_items = wcag_items.clone();
 		let children = all_items
 			.iter()
 			.filter_map(|item| {
-				(item.parent.id != AccessibleId::Null).then_some(item.object.id)
+				(item.parent.id != AccessibleId::Null).then_some(item.object.clone())
 			})
 			.collect();
-		let cache = Cache::new();
-		cache.add_all(all_items).await;
+		let cache = Cache::new(zbus_connection.clone());
+		cache.add_all(all_items);
 		(cache, children)
 	});
 	group.bench_function(BenchmarkId::new("traverse_cache", "zbus-docs"), |b| {
-		b.to_async(&rt).iter_batched(
+		b.iter_batched(
 			|| children.clone(),
-			|cs| async { traverse_cache(&cache, cs).await },
+			|cs| traverse_cache(&cache, cs),
 			BatchSize::SmallInput,
 		);
 	});
 
 	let mut rng = &mut rand::thread_rng();
-	let cache = Cache::new();
+	let cache = Cache::new(zbus_connection.clone());
 	let all_items = zbus_items.clone();
 	for size in [10, 100, 1000] {
 		let sample = all_items
 			.choose_multiple(&mut rng, size as usize)
-			.map(|item| item.object.id)
+			.map(|item| item.object.clone())
 			.collect::<Vec<_>>();
 		group.throughput(criterion::Throughput::Elements(size));
 		group.bench_function(BenchmarkId::new("reads_while_writing", size), |b| {
-			b.to_async(&rt).iter_batched(
+			b.iter_batched(
```

Comment:
Ah. I think this is the reason why there is such a drastic decrease in performance :laughing: I tried printing / logging to files from the `reads_while_writing` function, it doesn't appear to be getting executed. This needs to go back to `b.to_async(&rt)`.